sunflower-land/sunflower-land
# [FEAT] Add Dynamic Bulk Buy Button for Seeds

# Description

- Add a Dynamic Bulk Buy Button for Seeds. Amount to bulk buy dynamically changes depending on how close player is to hitting inventory limit
- Modified the Buy 10 Button to a static buy 10 button as the dynamic function is present in the new dynamic buy button as explained above and shown in the screenshots below. This button will disappear when stock<=10 OR difference between inventory amount and inventory limit <=10
- Added a message when bulk buy amount < stock to explain the bulk buy amount

- Bug Fix: Fix behaviour where shop doesn't stop sales at inventory limit, but stop sales when inventory is 1 above inventory limit. For example sunflower seed inventory limit is 1200, but you are able to buy at 1200 but not at 1201. 

## Screenshots
If amount to be bought doesn't go above inventory limit, buys all seeds available
![image](https://github.com/sunflower-land/sunflower-land/assets/101262042/03e60f8e-eec3-4348-a398-edf42615387f)

If amount to be bought goes above inventory limit, buys only the amount enough to hit inventory limit. Message appears when the amount to be bought is lower than remaining stock
![image](https://github.com/sunflower-land/sunflower-land/assets/101262042/3e5f3f42-b30f-4a23-8dae-d55f3a8ac2a9)

When Stock <= 10, Buy 10 Button disappears, bulk buy button adapts to remaining stock left (similar to before except the implementation is different)
![image](https://github.com/sunflower-land/sunflower-land/assets/101262042/ec7980b9-991e-44b0-9e9a-aa5333fdbf71)

When difference <= 10, Buy 10 Button disappears, bulk buy button adapts to remaining space to inventory limit
![image](https://github.com/sunflower-land/sunflower-land/assets/101262042/e3af89b3-fb42-4da8-8c2b-4c08234d0e6f)

Confirmation
![image](https://github.com/sunflower-land/sunflower-land/assets/101262042/9a9809e3-d18e-4784-ba22-f53038fc80c9)

Mobile View
![Screenshot_20240606_191219_Brave](https://github.com/sunflower-land/sunflower-land/assets/101262042/35a0b3b2-68b9-43d1-916b-ab4d2fc34ef4)


Fixes #issue

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?


- Have no seeds in inventory, Bulk Buy button should show all available stock
- Set inventory number such that the difference between inventory amount and inventory limit < remaining stock. Bulk Buy Button should show the difference
- Set remaining stock <= 10. Buy 10 button will disappear and bulk buy button will show the remaining stock
- Set inventory number such that the difference between inventory amount and inventory limit <= to 10. Buy 10 button will disappear and Bulk buy button will show the difference
- If Bulk buy button > 10 it will show a confirmation message. If bulk buy < 10 it will buy that amount without confirmation

# Checklist:
